js/sites.js

```diff
@@ -259,35 +285,22 @@
      */
     captureIfMissing() {
       if (!document.hidden && !this.link.imageURI) {
-        gNewTab.sendToBrowser("NewTab:BackgroundPageThumbs", {
-          url: this.url
+        gNewTab.sendToBrowser("NewTab:CaptureBackgroundPageThumbs", {
+          link: this.link,
         });
       }
     },
 
     /**
-     * Refreshes the thumbnail for the site.
-     */
-    refreshThumbnail() {
-      gNewTab.sendToBrowser("NewTab:PageThumbs", {
-        link: this.link
-      });
-    },
-
-    /**
-     * Render the correct thumbnail for the site.
-     *
-     * @param {Object} aData Contains enhanced links and the URI generated from the
-     *        current URL.
+     * Show the enhanced thumbnail.
      */
-    getURI(aData) {
+    showEnhancedThumbnail() {
       let thumbnail = this._querySelector(".newtab-thumbnail");
       if (this.link.bgColor) {
         thumbnail.style.backgroundColor = this.link.bgColor;
       }
 
-      let uri = this.link.imageURI || aData.uri;
-      thumbnail.style.backgroundImage = `url("${uri}")`;
+      thumbnail.style.background = `url("${this.link.imageURI}")`;
```

Comment:
in line 309 below, how does it make sense to have `if(this.link.type !=== this.link.type)`